src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -999,7 +999,7 @@ public HttpClientChannelAsyncRequest(HttpClientRequestChannel channel)
 
                         if (!suppressEntityBody)
                         {
-                            _httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
+                            httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
```

Comment:
Just wondering if line 981 should be changed to 
```
httpRequestMessage = _channel.GetHttpRequestMessage(_via);
```

Reply:
Unfortunately not really :(  _httpRequestMessage is used in a few other code paths that can run simultaneously (and like you probably suspect, that code can run into the same kind of race condition that we're fixing here). So this fix does not completely eliminate all the issues - it just mitigates the most frequent one.